op-node/rollup/derive/span_batch_test.go

```diff
@@ -602,3 +602,42 @@ func TestSpanBatchTotalBlockTxCountNotOverflow(t *testing.T) {
 
 	require.ErrorIs(t, err, ErrTooBigSpanBatchSize)
 }
+
+func TestSpanBatchRevert(t *testing.T) {
+	rng := rand.New(rand.NewSource(0x44337711))
+
+	chainID := new(big.Int).SetUint64(rng.Uint64())
+
+	singularBatches := RandomValidConsecutiveSingularBatches(rng, chainID)
+	// initialize empty span batch
+	spanBatch := initializedSpanBatch([]*SingularBatch{}, uint64(0), chainID)
+
+	for i := 0; i < len(singularBatches); i++ {
+		err := spanBatch.AppendSingularBatch(singularBatches[i], uint64(i))
+		require.NoError(t, err)
+		spanBatch.revertLastBatch()
+
+		spanBatch2 := initializedSpanBatch(singularBatches[:i], uint64(0), chainID)
```

Comment:
Rather than appending and removing the same batch (and then adding it back at the end of the loop), why not test as follows:

* `spanBatch1` is made of all batches [0:n]
* `spanBatch2` is made of batches [0:i], for each `i` from 0->n
* for each `i`, revert `spanBatch1` until it only contains batches [0:i]
* Test equality

This way you could confirm that reverting works to any depth. In the current test, it only adds a singular batch to be reverted on each loop, which is kinda the same test every time.

Reply:
The revert call is supposed to be paired with `AppendSingularBatch`, so it's only called after `AppendSingularBatch` is successfully called. I added a comment to clarify the usage.